### PR TITLE
tools: exit(1) when test/build fail

### PR DESCRIPTION
**Pull request by hawkingrei**
Signed-off-by: Weizhen Wang <wangweizhen@pingcap.com>

### What problem does this PR solve?

Issue Number: close #xxx

Issue Number: ref #30822

Problem Summary:

### What is changed and how it works?

It is friendly for CI to exit(1) when test/build fail.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Review by tiancaiamao — COMMENTED** [review]
Why not just

```
func main() {
     err := doSth()
     if err != nil {
        os.Exit(-1)
     }
}
```

**Comment by hawkingrei** [comment]
> Why not just
> 
> ```
> func main() {
>      err := doSth()
>      if err != nil {
>         os.Exit(-1)
>      }
> }
> ```

It is friendly that we can check the status when all the tasks finish running.

**Comment by hawkingrei** [comment]
We also can create an error to return.

**Comment by tiancaiamao** [comment]
1. It's unnecessary to introduce a external atomic package for such simple changes.
2. If you want to exit immediately after the first error, just `os.Exit()`, otherwise if you want to collect all the exit info, you can add a field in worker struct, here is `type numa struct {exitErr error ...}`

File: tools/check/ut.go [hunk 1]
```diff
@@ -341,14 +355,15 @@ type numa struct {
...
-func (n *numa) worker(wg *sync.WaitGroup, ch chan task) {
+func (n *numa) worker(wg *sync.WaitGroup, ch chan task, ErrCh chan struct{}) {
 	defer wg.Done()
 	for t := range ch {
 		start := time.Now()
 		res := n.runTestCase(t.pkg, t.test, t.old)
 		if res.err != nil {
 			fmt.Println("[FAIL] ", t.pkg, t.test, t.old, time.Since(start), res.err)
 			io.Copy(os.Stderr, &res.output)
+			ErrCh <- struct{}{}
```
Comment (tiancaiamao) [inline tools/check/ut.go:366]: This code is buggy, if there are several error send to the ErrCh, the code will block here forever

Comment (hawkingrei) [reply tools/check/ut.go]: Removed

File: tools/check/ut.go [hunk 2]
```diff
@@ -262,6 +266,12 @@ func cmdRun(args ...string) {
 	}
 	close(taskCh)
 	wg.Wait()
+	for _, work := range works {
+		if work.Fail {
+			os.Exit(1)
+			break
```
Comment (tangenta) [inline tools/check/ut.go:272]: This `break` is unreachable.

Comment (hawkingrei) [reply tools/check/ut.go]: Fixed.

**Review by tangenta — APPROVED** [review]

**Pull request merged**